### ISSUE TITLE
Invalid option '--ext' - perhaps you meant '-c'?

### ISSUE THREAD

**Issue by jimbojw** [eslint/eslint#16531]
### Environment

Node version: v16.18.1
npm version: 8.19.2
Local ESLint version: 8.27.0
Global ESLint version: N/A
Operating System: Mac OS X

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
N/A
```
</details>

```js
N/A
```

### What did you expect to happen?

Running `eslint --ext .ts` should find and lint TypeScript files.

### What actually happened?

Running `eslint --ext .ts` produces this error:

```
Invalid option '--ext' - perhaps you meant '-c'?
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

Reverting to v8.17.0 fixes the issue. Not sure which version broke this.

**Comment by einojokipii** [comment]
Doesn't the eslint command require some location to search for files with the given extension? e.g. to run

`eslint src --ext .ts` instead of just `eslint --ext .ts` which is what he tried?

**Comment by ljharb** [comment]
I think what you want is `eslint --ext=.ts .` so that it looks everywhere in the current directory (iow, the location to search, as @einojokipii said)

**Comment by mdjermanovic** [comment]
That's correct, the command should have at least one file/directory/pattern argument. Otherwise, it won't lint anything.

However, it shouldn't throw `Invalid option '--ext'` as in the original issue. This is most likely a problem with switching to the [flat config](https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new) mode, perhaps there's an `eslint.config.js` file in the same directory or in a parent directory.

**Comment by mdjermanovic** [comment]
No response, so closing.

**Comment by jimbojw** [comment]
This appears to be a conflict with my `eslint.config.js` file although the debug output makes no reference to this.

Output with `--debug`:

```sh
$ npx eslint --ext .ts --debug
  eslint:cli CLI args: [ '--ext', '.ts', '--debug' ] +0ms
  eslint:cli Using flat config? true +4ms
  eslint:cli Error parsing CLI options: Invalid option '--ext' - perhaps you meant '-c'? +6ms
Invalid option '--ext' - perhaps you meant '-c'?
```

Output with `--debug` AND `eslint.config.js` removed:

```sh
$ npx eslint --ext .ts --debug
  eslint:cli CLI args: [ '--ext', '.ts', '--debug' ] +0ms
  eslint:cli Using flat config? false +5ms
  eslint:cli Running on files +5ms
  eslintrc:config-array-factory Loading .eslintignore file: <path removed>/.eslintignore +0ms
  eslintrc:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/**/node_modules/*' ], basePath: '<path removed>', loose: false } ] +0ms
  eslintrc:ignore-pattern   processed: { basePath: '<path removed>', patterns: [ '/**/node_modules/*' ] } +1ms
  eslintrc:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/**/node_modules/*' ], basePath: '<path removed>', loose: false } ] +0ms
  eslintrc:ignore-pattern   processed: { basePath: '<path removed>', patterns: [ '/**/node_modules/*' ] } +0ms
  eslint:file-enumerator Start to iterate files: [] +0ms
  eslint:file-enumerator Complete iterating files: [] +0ms
  eslint:cli-engine Linting complete in: 0ms +0ms
```

Contents of `.eslintignore`:

```
dist
rollup.config.js
webpack.config.js
```

Contents of conflicting `eslint.config.js`:

```js
const fs = require('fs');
const skipWords = fs.readFileSync('eslint.skipWords.txt', 'utf8')
                      .split('\n')
                      .map(line => line.trim())
                      .filter(line => !!line);

const OFF = 0;
const WARN = 1;
const ERROR = 2;

module.exports = {
  root: true,
  parser: '@typescript-eslint/parser',
  plugins:
  [
    '@typescript-eslint',
    'spellcheck',
  ],
  extends:
  [
    'eslint:recommended',
    'plugin:@typescript-eslint/recommended',
    'plugin:@typescript-eslint/recommended-requiring-type-checking',
  ],
  parserOptions: {
    project: ['tsconfig.json'],
    tsconfigRootDir:__dirname,
  },
  settings: {
    "import/parsers": {
      "@typescript-eslint/parser": [".ts", ".tsx"]
    },
    "import/resolver": {
      "typescript": {}
    },
  },
  rules: {
    '@typescript-eslint/no-confusing-void-expression': ERROR,
    '@typescript-eslint/prefer-readonly': ERROR,
    'spellcheck/spell-checker': [WARN, {
      'comments': true,
      'strings': true,
      'identifiers': true,
      'templates': false,
      'lang': 'en_US',
      'skipWords': skipWords,
      'skipIfMatch': [
        'dat\\.GUI',
        'TODO\\s*\\([^\\)]+\\):',
      ],
    }],
  },
};
```

It looks like something in my `eslint.config.js` is incompatible with the latest version. When I try to run it, I get this:

```sh
$ npx eslint --config=eslint.config.js --debug ./src
  eslint:cli CLI args: [ '--config=eslint.config.js', '--debug', './src' ] +0ms
  eslint:cli Using flat config? true +4ms
  eslint:cli Running on files +4ms
  eslint:flat-eslint Override config file path is eslint.config.js +0ms
  eslint:flat-eslint Loading config from <path removed>/eslint.config.js +0ms
  eslint:flat-eslint Config file URL is file://<path removed>/eslint.config.js +0ms
  eslint:flat-eslint Deleting cache file at <path removed>/.eslintcache +11ms

Oops! Something went wrong! :(

ESLint: 8.28.0

You are linting "./src", but all of the files matching the glob pattern "./src" are ignored.

If you don't want to lint these files, remove the pattern "./src" from the list of arguments passed to ESLint.

If you do want to lint these files, try the following solutions:

* Check your .eslintignore file, or the eslintIgnore property in package.json, to ensure that the files are not configured to be ignored.
* Explicitly list the files from this glob that you'd like to lint on the command-line, rather than providing a glob as an argument.
```

**Comment by jimbojw** [comment]
Update: Adding `ESLINT_USE_FLAT_CONFIG=false` to my script fixed the issue:

```json
"scripts": {
  "lint:ts": "ESLINT_USE_FLAT_CONFIG=false eslint --config eslint.config.js src/",
  ...
```

**Comment by ricardo-reis-1970** [comment]
I'm having the same issue. I just found out that as soon as the file `eslint.config.js` is present, doing `npx eslint -h` will not print the `--ext` option.

I think there should be a more precise set of guidelines for migration. I have a React app set with the latest vite and template `react-swc`. This generates an `.eslintrc.cjs` and it's difficult to convert it to the new format when this is the first contact with ESLint.

**Comment by Rec0iL99** [comment]
@ricardo-reis-1970 this issue is closed and we don't track closed issues. If you have a question please open a [new discussion](https://github.com/eslint/eslint/discussions/new/choose) or want to report a bug please open a [new issue](https://github.com/eslint/eslint/issues/new/choose)
